pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,263 @@
+	Group:    "user.openshift.io",
+	Version:  "v1",
+	Resource: "users",
+}
+
+type UserSettingsHandler struct {
+	K8sProxyConfig      *proxy.Config
+	Client              *http.Client
+	Endpoint            string
+	ServiceAccountToken string
+}
+
+func (h *UserSettingsHandler) HandleUserSettings(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		configMap, err := h.getUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to get user settings: %v", err)})
+		} else {
+			serverutils.SendResponse(w, http.StatusOK, configMap)
+		}
+	case http.MethodPost:
+		configMap, err := h.createUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to create user settings: %v", err)})
+		} else {
+			serverutils.SendResponse(w, http.StatusOK, configMap)
+		}
+	case http.MethodDelete:
+		err := h.deleteUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to delete user settings: %v", err)})
+		} else {
+			w.WriteHeader(http.StatusNoContent)
+		}
+	default:
+		w.Header().Set("Allow", "GET, POST, DELETE")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Unsupported method, supported methods are GET POST DELETE"})
+	}
+}
+
+// Fetch the current user with auth.User token and returning the user-setting ConfigMap.
+func (h *UserSettingsHandler) getUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Get(context, resourceData.getConfigMapName(), meta.GetOptions{})
+}
+
+// Fetch the current user with auth.User token and create a new user-setting ConfigMap.
+// Returns the existing ConfigMap if it is already exist.
+func (h *UserSettingsHandler) createUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	role := &rbac.Role{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "Role",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleName(),
+		},
+		Rules: []rbac.PolicyRule{
+			rbac.PolicyRule{
+				APIGroups: []string{
+					"", // Core group, not "v1"
+				},
+				Resources: []string{
+					"configmaps", // Not "ConfigMap"
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"patch",
+					"update",
+					"watch",
+				},
+				ResourceNames: []string{
+					resourceData.getConfigMapName(),
+				},
+			},
+		},
+	}
+
+	roleBinding := &rbac.RoleBinding{
```

Comment:
Lets create a helper for this, so we can add unit-test.

Reply:
Done, see helpers.go